zkevm-circuits/src/pi_circuit.rs

```diff
@@ -386,6 +626,67 @@ impl<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize>
         Ok(())
     }
 
+    /// Assigns one calldata row
+    #[allow(clippy::too_many_arguments)]
+    fn assign_tx_calldata_row(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        tx_id: usize,
+        index: usize,
+        tx_value: F,
+        is_final: bool,
+        gas_cost: F,
+        raw_pi_vals: &mut [F],
+    ) -> Result<(), Error> {
+        let tx_id = F::from(tx_id as u64);
+        let tx_id_inv = tx_id.invert().unwrap_or(F::zero());
+        let tag = F::from(TxFieldTag::CallData as u64);
+        let index = F::from(index as u64);
+        let tx_value = tx_value;
+        let tx_value_inv = tx_value.invert().unwrap_or(F::zero());
+        let is_final = if is_final { F::one() } else { F::zero() };
+
+        self.q_tx_calldata.enable(region, offset)?;
+
+        // Assign vals to Tx_table
+        region.assign_advice(|| "tx_id", self.tx_id, offset, || Ok(tx_id))?;
+        region.assign_advice(|| "tx_id_inv", self.tx_id_inv, offset, || Ok(tx_id_inv))?;
+        region.assign_fixed(|| "tag", self.tag, offset, || Ok(tag))?;
+        region.assign_advice(|| "index", self.index, offset, || Ok(index))?;
+        region.assign_advice(|| "tx_value", self.tx_value, offset, || Ok(tx_value))?;
+        region.assign_advice(
+            || "tx_value_inv",
+            self.tx_value_inv,
+            offset,
+            || Ok(tx_value_inv),
+        )?;
+        region.assign_advice(|| "is_final", self.is_final, offset, || Ok(is_final))?;
+        region.assign_advice(
+            || "gas_cost",
+            self.calldata_gas_cost,
+            offset,
+            || Ok(gas_cost),
+        )?;
+
+        // Assign vals to raw_public_inputs column
+        let tx_table_len = TX_LEN * MAX_TXS + 1;
+
+        let value_offset = BLOCK_LEN + 1 + EXTRA_LEN + 2 * tx_table_len;
```

Comment:
Even though the offsets are computed correctly I think it would be a good idea to set `value_offset` to the start of the `CallDataBytes` portion of the `raw_public_inputs`. So having:
`let value_offset = BLOCK_LEN + 1 + EXTRA_LEN + 3 * tx_table_len;` and reseting `offset=0` when the assignment of CallDataBytes starts.

Reply:
yeah, it makes sense.